pages/index.js

```diff
@@ -78,61 +103,102 @@ export default function HomePage({ onAddEmotionEntry, handleToggleTooltip }) {
     setIsFormSubmitted(!isFormSubmitted);
   }
 
+  //logic for Graph
+  const currentShortDate = new Date().toISOString().slice(0, 10);
+  function compare(a, b) {
+    if (a.isoDate < b.isoDate) {
+      return -1;
+    }
+    if (a.isoDate > b.isoDate) {
+      return 1;
+    }
+    return 0;
+  }
+
+  const xValues = emotionEntries
+    .filter((entry) => currentShortDate === entry.isoDate?.slice(0, 10))
+    .sort(compare)
+    .map((entry) => entry.timeAndDate.slice(-5));
+
+  const yValues = emotionEntries
+    .filter((entry) => currentShortDate === entry.isoDate?.slice(0, 10))
+    .sort(compare)
+    .map((entry) => entry.tensionLevel);
```

Comment:
here you are performing double the work. Please filter and sort first and then split the values into xValues and yValues:

```js
const data = emotionEntries
    .filter((entry) => currentShortDate === entry.isoDate?.slice(0, 10))
    .sort(compare)
const xValues = data.map(entry => entry.timeAndDate.slice(-5))
const yValues = data.map((entry) => entry.tensionLevel);
```

Reply:
Hi, thanks for the review. Yeah, I had the feeling that bit of my code could have been better ... but the solution I had in mind was not as short as yours so I did not change it. But changed it now thanks to your suggestion.